_posts/2021-02-24-#21224.md

```diff
@@ -0,0 +1,71 @@
+---
+layout: pr
+date: 2021-02-24
+title: "Halt processing of unrequested transactions"
+pr: 21224
+authors: [ariard]
+components: ["p2p"]
+host: ellemouton
+status: upcoming
+commit:
+---
+
+## Notes
+
+- Transaction relay is a three step process: `inv` -> `getdata` -> `tx`:
+  1. the relaying node sends an `inv` message to the receiving node to announce
+     a new transaction.
+  2. the receiving node sends a `getdata` message to the relaying node to
+     request the full transaction.
+  3. the relaying node delivers a `tx` message to the receiving node. If the
+     relaying node is no longer able to deliver the transaction, it responds
+     with `notfound` instead of the `tx`.
+
+  You can learn more about these messages [here](https://en.bitcoin.it/wiki/Protocol_documentation#inv).
+
+- Currently a node will process a transaction that it receives even if it has not 
+  made a request for that transaction (i.e: If it recieves a `tx` message without
+  having first sent a `getdata` message asking for the `tx` message then it will
+  still process the transaction).
+
+- This PR aims to change that by requiring a transaction to first be requested
+  by a node before it is relayed to that node from one of its peers.
+
+- See the PR’s author's motivation behind the PR along with some of the
+  concerns that this change brings about:
+  [see mailing list thread here](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2021-February/018391.html)
+
+## Questions
+
+1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or
+   NACK?](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)
+
+2. What is the `TxRequestTracker` class used for? See [`src/txrequest.h`](https://github.com/bitcoin/bitcoin/blob/b805dbb0b9c90dadef0424e5b3bf86ac308e103e/src/txrequest.h)
+
+3. What are the different states that a peer/tx “announcement” can be in? Where
+   in the code is an announcement shifted between the different states?
+
+4. What does the author claim that the problem is with the current way in which 
+   transaction messages are handled?
+
+5. What is the change that this PR is making and how does it solve the problem
+   mentioned in Q5?
+
+6. There are a lot of switches on the `PR_RELAY` flag in code relevant to this
```

Comment:
I think you should raise up the question if this PR should introduce a new permission to unify tx-announcement related behaviors which could be  different from link-type authorizations. At least I'm waiting feedback on this.